spring-jms/src/test/java/org/springframework/jms/support/converter/MappingGsonMessageConverterTests.java

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright 2002-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.jms.support.converter;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.BDDMockito.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.isA;
+import static org.mockito.BDDMockito.mock;
+import static org.mockito.BDDMockito.verify;
+
+import java.io.ByteArrayInputStream;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.jms.BytesMessage;
+import javax.jms.JMSException;
+import javax.jms.Session;
+import javax.jms.TextMessage;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.springframework.core.MethodParameter;
+
+/**
+ * @author Arjen Poutsma
+ * @author Dave Syer
+ * @author Stephane Nicoll
+ * @author Marten Deinum
+ */
+public class MappingGsonMessageConverterTests {
+
+	@Rule
+	public final ExpectedException thrown = ExpectedException.none();
+
+	private MappingGsonMessageConverter converter;
+
+	private Session sessionMock;
+
+	@Before
+	public void setUp() throws Exception {
+		sessionMock = mock(Session.class);
+		converter = new MappingGsonMessageConverter();
+		converter.setEncodingPropertyName("__encoding__");
+		converter.setTypeIdPropertyName("__typeid__");
```

Comment:
I would pull both values into class level constants, because the type id property needs to match in the tests.